File: src/main/java/ee/taltech/gtm/listener/CustomEditorMouseListener.java
package ee.taltech.gtm.listener;

import com.intellij.openapi.editor.event.EditorMouseEvent;
import com.intellij.openapi.editor.event.EditorMouseListener;
import com.intellij.openapi.fileEditor.FileDocumentManager;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import ee.taltech.gtm.GtmWrapper;
import org.jetbrains.annotations.NotNull;

public class CustomEditorMouseListener implements EditorMouseListener {

    @Override
    public void mousePressed(@NotNull EditorMouseEvent event) {
        FileDocumentManager instance = FileDocumentManager.getInstance();
        VirtualFile file = instance.getFile(event.getEditor().getDocument());
        Project project = event.getEditor().getProject();
        GtmWrapper.getInstance().recordFile(project, file);
    }
}

File: src/main/java/ee/taltech/gtm/listener/CustomEditorMouseListener.kt
package ee.taltech.gtm.listener

import com.intellij.openapi.editor.event.EditorMouseEvent
import com.intellij.openapi.editor.event.EditorMouseListener
import com.intellij.openapi.fileEditor.FileDocumentManager
import ee.taltech.gtm.GtmWrapper

class CustomEditorMouseListener : EditorMouseListener {
    override fun mousePressed(event: EditorMouseEvent) {
        val instance = FileDocumentManager.getInstance()
        val file = instance.getFile(event.editor.document) ?: return
        val project = event.editor.project
        GtmWrapper.instance.recordFile(project, file)
    }
}